utils/librates.go

```diff
@@ -577,27 +711,34 @@ func (rIv *RateSInterval) CompressEquals(rIv2 *RateSInterval) (eq bool) {
 	return true
 }
 
-func (rIv *RateSInterval) Cost() *decimal.Big {
+func (rIv *RateSInterval) Cost(rts map[string]*IntervalRate) (cost *decimal.Big, err error) {
 	if rIv.cost == nil {
 		rIv.cost = new(decimal.Big)
 		for _, incrm := range rIv.Increments {
-			rIv.cost = SumBig(rIv.cost, incrm.Cost())
+			cstIncr, err := incrm.Cost(rts)
+			if err != nil {
+				return nil, err
+			}
+			rIv.cost = SumBig(rIv.cost, cstIncr)
 		}
 	}
-	return rIv.cost
+	return rIv.cost, nil
 }
 
 // CompressEquals compares two RateSIncrement for Compress function
 func (rIcr *RateSIncrement) CompressEquals(rIcr2 *RateSIncrement) (eq bool) {
-	return rIcr.Rate.UID() == rIcr2.Rate.UID() &&
+	return rIcr.RateID == rIcr2.RateID &&
 		rIcr.IntervalRateIndex == rIcr2.IntervalRateIndex &&
 		rIcr.Usage.Big.Cmp(rIcr2.Usage.Big) == 0
 }
 
 // Cost computes the Cost on RateSIncrement
-func (rIcr *RateSIncrement) Cost() *decimal.Big {
+func (rIcr *RateSIncrement) Cost(rts map[string]*IntervalRate) (cost *decimal.Big, err error) {
```

Comment:
Please remove the error here since panic should be enough in case of wrong populating of the map.